frontend/src/frontend/code_generator/user_callgraph.py

```diff
@@ -70,7 +76,41 @@ def format_vars_definition(self) -> str:
         return self.format_code_block_body(self.global_vars_def)
 
     def format_code_block_body(self, code_block: blocks.CodeBlock) -> str:
-        return code_block.get_instructions()
+        # code block body contains either instructions or a prefetch
+        cbb = code_block.code_block_body
+        try:
```

Comment:
It would be cleaner to just check directly if the code block body contains instructions or not, rather than catching the exception.

Reply:
IMO, asking for forgiveness rather than permission applies when there isn't an obvious way to know if your request will fail or not, or if it's an error you can anticipate but don't have much control over (e.g. OSError). In this case, it's pretty clear that an entirely different function should be called based on a condition we can easily check.

Reply:
I changed it to use conditionals.